src/Model/room.js

```diff
@@ -0,0 +1,13 @@
+const room = {
```

Comment:
Değişkenlerimize `const` diyerek değiştirilemez olduğunu kabul etsek de objelerde işler biraz farklı gidiyor. 

![models added by beyzaerkan · Pull Request #36 · anchovycation_pingu-api - Google Chrome 29 07 2022 20_40_16](https://user-images.githubusercontent.com/56413673/181818974-dbf26bff-c17b-401f-be98-8cd8eeb2fc6e.png)

Örnekte de görülebileceği gibi objelerde `=` ile doğrudan atama yapılırken hata alınsa da sahip olduğu key'in değeri değiştirildiğinde hata verilmez çünkü ne değişkenin referansı değişir ne de doğrudan tipi. Bundan dolayı yaptığımız işlerin daha önce Metronom'da da yaşadığımız şema sorunu gibi diğer işlerimizi etkilememesi için doğrudan nesnenin aslını değil onun derin bir klonunu vermeliyiz. Bunun için şöyle bir kod kullanabiliriz.

```js
// src/Model/index.js
import room from './Room.js';
// Tüm modeller alınır.

const Models = {
  room,
  // ...
}

const createModelInstanceFrom = ( modelName) => {
  const model = Models[modelName];
  if(!model){
    throw new Error(`${modelName} adlı model bulunamadı!`);
  }
  return { ...model }; // Burada modelin tüm keyleri parçalanır ve onlardan yeni bir obje üretilir.
};

export default {
  ...Models,
  createModelInstanceFrom,
}

// src/Services/Room/index.js
import { createModelInstanceFrom } from 'src/Models';

const roomModel = new Model(createModelInstanceFrom('room'), 'rooms', { keyUnique: 'id' });
```

Detaylı bilgi için bkz.: **shallow copy** vs **deep copy**